### PR TITLE
v0.5.0

### PR DESCRIPTION
**Pull request by SD10**

### CODE REVIEW

**Comment by MacMeDan** [comment]
@SD10 Do we want to do a review on this as a whole?

**Comment by SD10** [comment]
@MacMeDan Yeah let's wait until we're ready to do a `v0.5.0` release. Unless something is raising a significant red flag before then.

**Comment by SD10** [comment]
The detected text doesn't even have gesture recognition... yet. 
What was I thinking moving this out from WIP? 😂

**Comment by SD10** [comment]
@MacMeDan I know you haven't had time to review this but I'm going to move forward to `v0.6.0`. I want to get out of pre-release by the end of the month. I need to move quicker 😅

File: Sources/MessageLabel.swift [hunk 1]
```diff
@@ -280,31 +293,132 @@ open class MessageLabel: UILabel {
...
+            return layoutManager.characterIndexForGlyph(at: glyphIndex)
+        } else {
+            return nil
+        }
+        
+    }
+
+    private func setupGestureRecognizers() {
+
+        print("Setting up gesture recognizers")
```
Comment (MacMeDan) [inline Sources/MessageLabel.swift]: We probably don't want to leave `print` statements in the final code that will clutter up the command line. These will build up in time and if it is working provide no value.

File: Sources/MessageLabel.swift [hunk 2]
```diff
@@ -280,31 +293,132 @@ open class MessageLabel: UILabel {
...
+        let longPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(handleGesture(_:)))
+        addGestureRecognizer(longPressGesture)
+        tapGesture.delegate = self
+
+        isUserInteractionEnabled = true
+    }
+
+    func handleGesture(_ gesture: UIGestureRecognizer) {
+
+        print("Handling gesture")
```
Comment (MacMeDan) [inline Sources/MessageLabel.swift]: remove print statement.

File: Sources/MessageLabel.swift [hunk 3]
```diff
@@ -280,31 +293,132 @@ open class MessageLabel: UILabel {
...
+                    handleGesture(for: detectorType, text: text)
+
+                }
+            }
+        }
+    }
+
+    private func handleGesture(for detectorType: DetectorType, text: NSAttributedString) {
+
+        print("Detector routing")
```
Comment (MacMeDan) [inline Sources/MessageLabel.swift]: remove print statement.

File: Sources/MessagesViewController.swift [hunk 4]
```diff
@@ -155,14 +155,22 @@ extension MessagesViewController: UICollectionViewDataSource {
...
+        let message = messagesDataSource.messageForItem(at: indexPath, in: messagesCollectionView)
+        cell.configure(with: message)
+
+        if let cellDelegate = messagesCollectionView.messageCellDelegate {
+            cell.delegate = cellDelegate
+        }
+
+        if let messageLabelDelegate = messagesCollectionView.messageLabelDelegate {
+            cell.messageLabel.delegate = messageLabelDelegate
+            print("Setting delegate")
```
Comment (MacMeDan) [inline Sources/MessagesViewController.swift]: remove print statement.

**Review by MacMeDan — COMMENTED** [review]
It would be nice if we could remove any extra `print` statements to keep the command line clean while building.

**Pull request merged**